flight/flight.py

```diff
@@ -0,0 +1,232 @@
+"""Initialization Code for Flight State Machine"""
+import asyncio
+import logging
+from mavsdk import System
+import mavsdk
+from multiprocessing import Queue
+
+import logger
+from communication import Communication
+from flight import config
+from flight.states import STATES, State
+from flight.state_settings import StateSettings
+
+SIM_ADDR: str = "udp://:14540"  # Address to connect to drone simulator
+CON_ADDR: str = "serial:///dev/ttyUSB0:921600"  # Address to connect to pixhawk w/ baud rate
+
+
+class DroneNotFoundError(Exception):
+    """
+    Exception for when the drone cannot connect
+    """
+
+    pass
+
+
+class StateMachine:
+    """
+    Class for the flight state machine
+
+    Attributes
+    ----------
+    current_state : State
+        State currently being run in the state machine
+
+    Methods
+    -------
+    run() -> None
+        Initiates the run method for each state in the state machine
+    """
+
+    def __init__(self, initial_state: State, drone: System) -> None:
+        """
+        Initializes flight state machine
+
+        Parameters
+        ----------
+        initial_state : State
+            First state of the flight state machine listed in STATES object
+        drone : System
+            MAVSDK object for drone control
+        """
+        self.current_state: State = initial_state
+        self.drone: System = drone
+
+    async def run(self) -> None:
+        """
+        Runs the flight code specific to each state until completion
+        """
+        while self.current_state:
+            self.current_state = await self.current_state.run(self.drone)  # type: ignore[assignment]
+
+
+async def log_flight_mode(drone: System) -> None:
+    """
+    Logs the drone's current flight mode
+
+    Parameters
+    ----------
+    drone : System
+        MAVSDK object for drone control
+    """
+    previous_flight_mode: str = ""
+    async for flight_mode in drone.telemetry.flight_mode():
```

Comment:
Should add a type hint for `flight_mode`.
```suggestion
    previous_flight_mode: str = ""
    flight_mode: str
    async for flight_mode in drone.telemetry.flight_mode():
```